pyccel/codegen/printing/cwrappercode.py

```diff
@@ -578,11 +584,12 @@ def _print_Interface(self, expr):
             mini_wrapper_func_body = []
             res_args = []
             mini_wrapper_func_vars = {a.name : a for a in func.arguments}
+            local_arg_vars = [a.clone(a.name, is_pointer=True) if hasattr(a, "rank") and a.rank > 0 else a for a in func.arguments]
```

Comment:
It would be better to check if `isinstance(a, Variable)` instead of `hasattr(a, "rank")` that way you check the existence of `a.rank` and `a.clone`

Reply:
is there no way to change this from the sematic stage when we create the `functionDef` node?

Reply:
the cwrapper doesn't go through the semantic stage.

Reply:
> is there no way to change this from the sematic stage when we create the functionDef node?

The same `FunctionDef` is fed to `cwrappercode` and to `ccode`/`fcode`. `ccode`/`fcode` prints it in the expected way, while `cwrappercode` uses the information in the `FunctionDef`'s signature to print a wrapper. If we change the `FunctionDef` itself then we will also change how we print the code in c or fortran which is not what we want